packages/nativescript-appdynamics/common.ts

```diff
@@ -16,10 +18,86 @@ export enum LoggingLevel {
 // the app key can be created from the AppDynamics control panel when adding a new app.
 // The app key should be of format "AD-AAA-BBB", AppDynamics will error if it's malformed.
 export interface AppdynamicsConfiguration {
+  /**
+   * Sets the application key used by the SDK.  This is a required property.
+   */
+  /**
+   * Sets the application key used by the SDK.  This is a required property.
+   */
   appKey: string;
-  collectorURL: string; // should be the 'EUM Cloud' / 'EUM Collector' address
-  screenshotURL: string; // should be the 'Screenshot Service' address
+
+  /**
+   * Sets the name of this mobile application.  If not set, the application name is the
+   * package name from the Android Manifest.  Most users will not need this.
+   *
+   * The `applicationName` may contain uppercase or lowercase letters ('A' through 'Z'), numbers,
+   * and underscores ('_').
+   *
+   * **NOTE:** If this property is set, all data reported from this application is associated with `applicationName`,
+   * and appears together in dashboards.
+   */
+  applicationName?: string;
+
+  /**
+   * Boolean value that indicates if automatic instrumentation is enabled.
+   *
+   * Setting this to `false` will disable all automatic instrumentation regardless of any code injection.
+   * The following features will also be effected:
+   *
+   * 1. {@link interactionCaptureMode} will be set to {@link InteractionCaptureMode.None}
+   * 2. Automatic Network requests from supported libraries will not be tracked.
+   * 3. Any UI transition events will not be tracked.
+   * 4. Screenshots (both automatic and manual) will be disabled.
+   *
+   * Set to `false` to disable automatic instrumentation, `true` to enable (default is enabled)
+   */
+  autoInstrument?: boolean;
+
+  /**
+   * Sets the URL of the collector to which the agent will report.
+   *
+   * This is NOT your controller URL. You likely do not need to call this method
+   * unless you have an on-premise EUM Processor.
+   */
+  collectorURL?: string; // should be the 'EUM Cloud' / 'EUM Collector' address
+  /**
+   * Sets the URL of the screenshot service to which the agent will upload screenshots.
+   *
+   * This is NOT your controller URL. You likely do not need to call this method
+   * unless you have an AppDynamics managed private cloud (very rare).
+   *
+   * **NOTE:** If you have an on-premise EUM Processor
+   *           and set the collector URL in {@link collectorURL},
+   *           then you do not need to call this method
+   *           because the two URLs are the same,
+   *           and the agent assumes that is the case.
+   */
+  screenshotURL?: string; // should be the 'Screenshot Service' address
+  /**
+   * Sets the logging level of the agent. Default is {@link LoggingLevel.LOGGING_LEVEL_NONE}.
+   *
+   * **WARNING:** Not recommended for production use.
+   */
   loggingLevel?: LoggingLevel; // defaults to 'Error'
```

Comment:
The existing inline comment and newly added block-level comment seem to conflict on what the default logging level is.